src/core/cuda_common.cpp

```diff
@@ -0,0 +1,52 @@
+#include "cuda_common.h"
+
+
+
+
+
+
```

Comment:
Missing comments for argument and return value. Also please try to be more detailed with the description.